src/util/MmapVectorImpl.h

```diff
@@ -35,19 +41,35 @@ void MmapVector<T>::writeMetaDataToEnd() {
   ofs.write((char*)&_size, sizeof(_size));
   ofs.write((char*)&_capacity, sizeof(_capacity));
   ofs.write((char*)&_bytesize, sizeof(_bytesize));
+  ofs.write((char*)&MagicNumber, sizeof(MagicNumber));
```

Comment:
since magic number is larger than an `unit8_t` this means the written value will depend on the endianess. But I think in this case this might not be bad since we can't read other-endianess data anyway since all of it is written in the non--endianess-portable `write(&data, sizeof(data));` way.